staging/src/k8s.io/apimachinery/pkg/runtime/converter.go

```diff
@@ -317,80 +375,144 @@ func sliceFromUnstructured(sv, dv reflect.Value) error {
 		if len(sv.Interface().(string)) > 0 {
 			marshalled, err := json.Marshal(sv.Interface())
 			if err != nil {
-				return fmt.Errorf("error encoding %s to json: %v", st, err)
+				return nil, fmt.Errorf("error encoding %s to json: %v", st, err)
 			}
 			// TODO: Is this Unmarshal needed?
 			var data []byte
 			err = json.Unmarshal(marshalled, &data)
 			if err != nil {
-				return fmt.Errorf("error decoding from json: %v", err)
+				return nil, fmt.Errorf("error decoding from json: %v", err)
 			}
 			dv.SetBytes(data)
 		} else {
 			dv.Set(reflect.Zero(dt))
 		}
-		return nil
+		return nil, nil
 	}
 	if st.Kind() != reflect.Slice {
-		return fmt.Errorf("cannot restore slice from %v", st.Kind())
+		return nil, fmt.Errorf("cannot restore slice from %v", st.Kind())
 	}
 
 	if sv.IsNil() {
 		dv.Set(reflect.Zero(dt))
-		return nil
+		return nil, nil
 	}
 	dv.Set(reflect.MakeSlice(dt, sv.Len(), sv.Cap()))
 	for i := 0; i < sv.Len(); i++ {
-		if err := fromUnstructured(sv.Index(i), dv.Index(i)); err != nil {
-			return err
+		strictErrs, err := fromUnstructured(sv.Index(i), dv.Index(i), fromUnstructuredOptions{
+			isInlined:           opts.isInlined,
+			matchedKeys:         opts.matchedKeys,
+			strictErrs:          opts.strictErrs,
+			parentPath:          append(opts.parentPath, "[", strconv.Itoa(i), "]"),
+			validationDirective: opts.validationDirective,
+		})
+		if err != nil {
+			return nil, err
+		}
+		if strictErrs != nil {
+			return strictErrs, nil
 		}
```

Comment:
we should accumulate strictErrs, not short circuit the `for i := 0; i < sv.Len(); i++ {` loop. need a test that would catch this short-circuit